agent/utils.mjs

```diff
@@ -0,0 +1,659 @@
+    return arrBytes;
+}
+
+export function bytesToHex( arrBytes, isInversiveOrder ) { // convert a byte array to a hex string
+    isInversiveOrder = !!(
+        ( isInversiveOrder != null && isInversiveOrder != undefined && isInversiveOrder )
+    );
+    const hex = [];
+    for( let i = 0; i < arrBytes.length; i++ ) {
+        const current = arrBytes[i] < 0 ? arrBytes[i] + 256 : arrBytes[i];
+        const c0 = ( current >>> 4 ).toString( 16 );
+        const c1 = ( current & 0xF ).toString( 16 );
+        if( isInversiveOrder ) {
+            hex.splice( 0, 0, c0 );
+            hex.splice( 1, 0, c1 );
+        } else {
+            hex.push( c0 );
+            hex.push( c1 );
+        }
+    }
+    return hex.join( "" );
+}
+
+export function bytesAlignLeftWithZeroes( arrBytes, cntMin ) {
+    const arrOneZeroByte = new Uint8Array( 1 );
+    arrOneZeroByte[0] = 0;
+    while( arrBytes.length < cntMin )
+        arrBytes = bytesConcat( arrOneZeroByte, arrBytes );
+    return arrBytes;
+}
+
+export function bytesAlignRightWithZeroes( arrBytes, cntMin ) {
+    const arrOneZeroByte = new Uint8Array( 1 );
+    arrOneZeroByte[0] = 0;
+    while( arrBytes.length < cntMin )
+        arrBytes = bytesConcat( arrBytes, arrOneZeroByte );
+    return arrBytes;
+}
+
+export function concatTypedArrays( a, b ) { // a, b TypedArray of same type
+    if( typeof a == "string" )
+        a = hexToBytes( a );
+    if( typeof b == "string" )
+        b = hexToBytes( b );
+    const c = new ( a.constructor )( a.length + b.length );
+    c.set( a, 0 );
+    c.set( b, a.length );
+    return c;
+}
+
+export function concatByte( ui8a, byte ) {
+    const b = new Uint8Array( 1 );
+    b[0] = byte;
+    return concatTypedArrays( ui8a, b );
+}
+
+export function bytesConcat( a1, a2 ) {
+    a1 = a1 || new Uint8Array();
+    a2 = a2 || new Uint8Array();
+    return concatTypedArrays( a1, a2 );
+}
+
+export function toArrayBuffer( buf ) {
+    // see
+    // https://stackoverflow.com/questions/8609289
+    //          /convert-a-binary-nodejs-buffer-to-javascript-arraybuffer
+    const ab = new ArrayBuffer( buf.length );
+    const view = new Uint8Array( ab );
+    for( let i = 0; i < buf.length; ++i )
+        view[i] = buf[i];
+    return ab;
+}
+
+export function toBuffer( ab ) {
+    const buf = Buffer.alloc( ab.byteLength );
+    const view = new Uint8Array( ab );
+    for( let i = 0; i < buf.length; ++i )
+        buf[i] = view[i];
+    return buf;
+}
+
+export function invertArrayItemsLR( arr ) {
+    let i; const cnt = arr.length / 2;
+    for( i = 0; i < cnt; ++i ) {
+        const e1 = arr[i];
+        const e2 = arr[arr.length - i - 1];
+        arr[i] = e2;
+        arr[arr.length - i - 1] = e1;
+    }
+    return arr;
+}
+
+// see: https://developer.chrome.com/blog/how-to-convert-arraybuffer-to-and-from-string/
+export function ab2str( buf ) {
+    return String.fromCharCode.apply( null, new Uint16Array( buf ) );
+}
+
+export function str2ab( str ) {
```

Comment:
I would suggest to use TextEncoder and then uint8buffer to array buffer
https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder

Reply:
Not needed at all. Both `ab2str` and `str2ab` are unused. Will remove them. Thanks.